src/v/cluster/metadata_dissemination_service.cc

```diff
@@ -136,6 +135,7 @@ ss::future<> metadata_dissemination_service::start() {
           return update_metadata_with_retries(std::move(addresses));
       });
 
+    _dispatch_timer.arm(_dissemination_interval);
```

Comment:
I recognize that I had mentioned that the race should be explained in the commit message, but do you think that will be enough to prevent the race from reappearing during refactoring?

To clarify: if there is a reasonable expectation that the starting order should not matter (and that does appear to be the case since we are adding _exceptions_ like this) then consulting the commit log may not be natural and a comment that is placed in such a way that requires it to be read during a refactor might be more useful.

Reply:
I've added the comment to commit message.

I think that the order is now natural and will not cause any issues during refactoring. We have a comment explaining dependency between dissemination service and controller in the `application.cc`. Starting timer in constructor is not common.